src/Rings/MPolyQuo.jl

```
@@ -1785,7 +1785,10 @@ julia> dim(a)
 0
 ```
 """
-function dim(a::MPolyQuoIdeal)
+dim(a::MPolyQuoIdeal) = krull_dim(a)
+
+function krull_dim(a::MPolyQuoIdeal)
+  @req is_noetherian(base_ring(a)) "Krull dimension is only supported for Noetherian rings."
   if a.dim === nothing 
     a.dim = Singular.dimension(singular_groebner_generators(a))
     a.dim == -1 && (a.dim = -inf)
```

So we allow returning `-inf`. For the algebraic geometry group, is that for trivial rings (i.e. `a` being the 1-ideal) ? If so, we could of course also define `krull_dim` for trivial rings (right now the methods in AA will throw an error in that case).

Both have pros and cons. I am neutral on this in the end but feel we should decide on one (`krull_dim` for trivial rings should either throw an error, or return `-inf`). 

We can discuss it during triage but @oscar-system/algebraic-geometry and @thofma should be aware.

---

A similar discussion first arose in the context of `vector_space_dimension` where it turned out after a while that we needed to catch this case to allow higher functionality to robustly handle the 0-ring and the 0-module. 

The background is that such rings might legitimately arise as results of intermediate computations and the corner cases need to be recognized and handled appropriately. An error was simply not specific and flexible enough in that case. I could well imagine that this is similar here.

---

I agree.

---

@simonbrandhorst and myself also agree: `-1` should not be a valid return value for krull dimension. We also propose `-Inf`.

---

Just to be clear: $-1$ never was an acceptable proposal. The choice was between:
1. have `krull_dim` return `-inf` for trivial rings
2. have `krull_dim` throw an error for trivial rings

Approach 1 is more mathematical but has the drawback that one has to be careful to avoid type stability. E.g. this would be type unstable:
```julia
  ...
  d = krull_dim(myobj)
  d == -inf && return 0  # handle trivial case
  if d > 5
     ...
```
To make it type stable one must change the `==` to `===` i.e. `d === -inf`.

Approach 2 is less mathematical but convenient for programmers. There one would write the code like this and it would automatically be type stable.
```julia
  ...
  is_trivial(myobj) && return 0  # handle trivial case
  d = krull_dim(myobj)
  if d > 5
     ...
```

---

We should stay consistent with decisions for other dimension commands: `-inf` has already been established for `vector_space_dimesion` (after a lengthy discussion and a previous decision for an error, which caused more problems than it did good) and follows the more mathematical perspective. So I still favor for option 1. 

Of course, I meant `inf`and not `-inf`, but the content should still be clear....

---

If we go into that direction we should also do https://github.com/Nemocas/Nemo.jl/pull/2116

---

That said, I don't understand what you mean by
> `-inf` has already been established for `vector_space_dimension`

In which sense has it been established, and where and by whom? Surely a vector space always has dimension $\geq 0$ ?!

Also, I find it interesting that for infinite-dimensional vector spaces, you have decided to throw an error instead of returning `inf`, which seems inconsistent again? I.e.:
```julia
function vector_space_dimension(A::MPolyQuoRing)
  if !isa(coefficient_ring(A), AbstractAlgebra.Field)
    error("vector_space_dimension requires a coefficient ring that is a field")
  end
  is_finite_dimensional_vector_space(A) || throw(InfiniteDimensionError())
...
```
Note that
```julia
function is_finite_dimensional_vector_space(A::MPolyQuoRing)
  # We check '<=' because A might be the zero ring, so dim(A) == -1
  return dim(A) <= 0
end
```
which could also be implemented as
```
function is_finite_dimensional_vector_space(A::MPolyQuoRing)
  return is_trivial(A) || krull_dim(A) == 0
end
```
(BTW regardless of what we do, I just noticed that function has a mistake in its docstring, it says:
```
`A` is finite-dimensional as a `K`-vector space iff it has Krull dimension zero.
```
Moreover, `is_finite_dimensional_vector_space` does not check if `MPolyQuoRing` is even defined over a field?

---

Yes, that was too quickly written typed on the cell phone without enough thinking. I meant: If the Krull-dimension is larger than 0,then vector_space_dimension should return `inf`.

---

I moved the rest of this into an issue, see https://github.com/oscar-system/Oscar.jl/issues/5056